mardi_importer/mardi_importer/zenodo/ZenodoSource.py

```diff
@@ -0,0 +1,249 @@
+            new_entry["title"] = entry.get("title")
+            new_entry["doi"] = entry.get("doi") 
+            # retrieve list of authors
+            # new_entry["authors"] = []
+            # for auth in entry_metadata.get("creators"):
+            #     new_entry["authors"].append(auth.get("name"))
+            # new_entry["description"] = entry_metadata.get("description")
+            new_entry["publication_date"] = entry_metadata.get("publication_date")
+            # new_entry["journal"] = entry_metadata.get("journal")
+            # new_entry["language"] = entry_metadata.get("language")
+            
+            # new_entry["license"] = entry_metadata.get("license")
+            # new_entry["instance_of"] = entry_metadata.get("resource_type").get("type")
+            # # retrieve list of community ids
+            # new_entry["communities"] = []
+            # if entry_metadata.get("communities") is not None:
+            #     for c in entry_metadada.get("communities"):
+            #         new_entry["communities"].append(c.get("id"))
+            new_entry["metadata"] = entry_metadata
+        
+        return new_entry
+
+    def pull(self):
+        #self.write_data_dump()
+        #self.process_data()
+        print("skip")
+
+    def create_local_entities(self):
+        filename = self.filepath + "/new_entities.json"
+        f = open(filename)
+        entities = json.load(f)
+
+        for prop_element in entities["properties"]:
+            prop = self.integrator.property.new()
+            prop.labels.set(language="en", value=prop_element["label"])
+            prop.descriptions.set(language="en", value=prop_element["description"])
+            prop.datatype = prop_element["datatype"]
+            if not prop.exists():
+                prop.write()
+
+        for item_element in entities["items"]:
+            item = self.integrator.item.new()
+            item.labels.set(language="en", value=item_element["label"])
+            item.descriptions.set(language="en", value=item_element["description"])
+            if "claims" in item_element:
+                for key, value in item_element["claims"].items():
+                    item.add_claim(key, value=value)
+            if not item.exists():
+                item.write()
+
+        
+
+    def push(self):
+
+        file = open(self.processed_dump_path + "/zenodoData_dict.pkl" , 'rb')
+        records_all = pickle.load(file)
+        file.close()
+
+        for x in records_all:
+            
+            entry = ZenodoResource.ZenodoResource(
+                self.integrator,
+                zenodo_id = str(x),
+                #title = records_all[x]['title'],
+            #_publication_date = entry['publication_date'],
+            #_authors = entry['authors'],
+            #_resource_type = entry['resource_type'],
+            #_license = entry['lisence'],
+            metadata = records_all[x]['metadata'])
```

Comment:
I think this metadata should be deleted. Whatever is passed in this initialization is immediately overwritten in  https://github.com/MaRDI4NFDI/docker-importer/blob/537a54c4657c2e310159b551e74580748ceecc53/mardi_importer/mardi_importer/publications/ZenodoResource.py#L27 once the ZenodoResource gets created.

Maybe it would be enough to loop through the list of zenodo_ids, which can be saved directly as an attribute of the class. Then one could get rid of all the json, pkl files. Just a suggestion.